### PR TITLE
Report send api call success

### PR DESCRIPTION
**Pull request by chankiyu22**

### CODE REVIEW

File: pkg/lib/infra/sms/accessyou/apis/sendsms.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		"POST",
+		fmt.Sprintf(
+			"%v/sendsms.php?accountno=%v&pwd=%v&tid=1&phone=%v&a=%v&user=%v&from=%v",
+			baseUrl,
+			accountNo,
+			pwd,
+			to,
+			body,
+			user,
+			sender),
```
Comment (louischan-oursky) [inline pkg/lib/infra/sms/accessyou/apis/sendsms.go:32]: You construct URL with url.URL and url.Values, not by string concatenation and printf.

File: pkg/lib/infra/sms/accessyou/apis/sendsms.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	}
+	defer resp.Body.Close()
+
+	respData, err := io.ReadAll(resp.Body)
+
+	// The response data is in format
+	// "\ufeff{\"msg_status\":\"100\",\"msg_status_desc\":\"Successfully submitted message. \\u6267\\u884c\\u6210\\u529f\",\"phoneno\":\"852********\",\"msg_id\":852309279}"
+
+	// Remove BOM token from resp json
+	respData = bytes.Replace(respData, []byte("\ufeff"), []byte(""), -1)
```
Comment (louischan-oursky) [inline pkg/lib/infra/sms/accessyou/apis/sendsms.go:48]: Only remove leading BOM(s). You are replacing all BOM in the bytes here.

File: pkg/lib/infra/sms/sendcloud/models/send.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+func (r *SendRequest) Presign() string {
+	vars, _ := json.Marshal(r.vars)
+	return strings.Join([]string{
+		fmt.Sprintf("msgType=%v", r.msgType),
+		fmt.Sprintf("phone=%v", strings.Join(r.phone, ",")),
+		fmt.Sprintf("sendRequestId=%v", r.sendRequestId),
+		fmt.Sprintf("smsUser=%v", r.smsUser),
+		fmt.Sprintf("templateId=%v", r.templateId),
+		fmt.Sprintf("vars=%v", string(vars)),
+	}, "&")
```
Comment (louischan-oursky) [inline pkg/lib/infra/sms/sendcloud/models/send.go:57]: Not sure if the signature requires a specific order of the arguments.
Not sure if the signature requires you to use a x-www-form-urlencoded, if yes, then the value should be encoded with url.QueryEscape

Comment (chankiyu22) [reply pkg/lib/infra/sms/sendcloud/models/send.go:57]: The arguments need to be alphabetical order

https://www.sendcloud.net/doc/sms/
> 1. 将实际调用API的参数以字母升序(A-Z)排列, 不包括 smsKey 和 signature 字段本身

The values in signature not need to be url encoded.

https://www.sendcloud.net/doc/sms/
> 1. 生成签名时, 参数不要使用 'urlencode'. 在调用 api 时, 才需要对参数做 'urlencode'

Comment (louischan-oursky) [reply pkg/lib/infra/sms/sendcloud/models/send.go:57]: Then you should include these in the comments. For a reader like me, I have no clue what you write is intended or not.

Comment (chankiyu22) [reply pkg/lib/infra/sms/sendcloud/models/send.go:57]: Comments added via 5e2df93

**Pull request merged**